### DISCUSSION TITLE
Flashing ESP32 Wrover E with Gitpod

### DISCUSSION THREAD

**Discussion by groovetronic911** [arendst/Tasmota#18074] [Q&A]
Hello everyone,

I am new at flashing microcontrollers and I read a lot, but I have a problem i can't get solved.

I would like to flash my [Wrover E](https://de.aliexpress.com/item/1005004968903107.html?spm=a2g0o.productlist.main.1.73bcatmyatmyXN&algo_pvid=ad918767-82dd-4f9b-b06c-288eb32b18c1&algo_exp_id=ad918767-82dd-4f9b-b06c-288eb32b18c1-0&pdp_ext_f=%7B%22sku_id%22%3A%2212000031191261090%22%7D&pdp_npi=3%40dis%21EUR%215.37%215.37%21%21%21%21%21%40210217c716775894544855068d0750%2112000031191261090%21sea%21DE%211641688759&curPageLogUid=sQaVJSmCaC7P) with Tasmota:

What is necessary? Support for the included cam, BME280 and german language.

What did I do so far?

1. https://github.com/arendst/Tasmota/tree/master opended with Gitpod-Browser-Plugin.
2. --> tasmota/my_user_config.h
3. erased // at #define USE_BMP , #define USE_WEBCAM  and #define MY_LANGUAGE            de_DE
4. platformio run -e tasmota32

Wifi is available, i got data from BME280 and it's german - but no picture from the cam (ESP32-DevKit 1).
I also tried with the Blakadder-Template: {"NAME":"Freenove-WROVER-Cam","GPIO":[1,1,1,1,5088,5089,0,0,1,1,1,1,1,1,5090,5091,1,4992,5184,5152,1,5120,5024,5056,0,0,0,0,1,1,5094,5095,5092,1,1,5093],"FLAG":0,"BASE":1}
Where is my fault? What did i miss? What is also necessary to do?

Thank you so much for your help.
![tasmota](https://user-images.githubusercontent.com/126232265/221867773-2043bfa8-ff46-487e-9c36-21d8b177e610.png)

**Comment by Jason2866** [comment]
Webcam is a special Tasmota variant. You have to use the env `tasmota32-webcam` for compiling.
It is NOT recommended to add additional stuff (=drivers) to this build. Many are not possible to add, we restricted this in the setup. The esp32 MCU is on its limits doing the Webcam job...

**Comment by groovetronic911** [comment]
So the download of tasmota32-webcam.bin should be all? But I am not able to connect to its own WIFI.

If i compile a new one, what do I have to do?
Only take german language or do i have to remove // also at #define USE_WEBCAM and then "platformio run -e tasmota32-webcam"? Or only german language an then platformio run -e tasmota32-webcam ?

**Comment by Jason2866** [comment]
Place this env in file `platformio_tasmota_cenv.ini` to compile Webcam variant in DE
```
[env:tasmota32-DE-webcam]
extends                 = env:tasmota32_base
board                   = esp32-fix
board_build.f_cpu       = 240000000L
build_flags             = ${env:tasmota32_base.build_flags}
                          -DMY_LANGUAGE=de_DE
                          -DFIRMWARE_WEBCAM
                          -DCAMERA_MODEL_AI_THINKER
                          -DOTA_URL='""'
lib_extra_dirs          = lib/lib_ssl, lib/libesp32
```

`pio run -e tasmota32-DE-webcam`

**Comment by groovetronic911** [comment]
Unfortunately - i got no picture.

Now I ordered a ESP-Cam-Module and i will try again...perhaps the cam is damaged.

Thank you for your help so far! :-)

**Comment by Jason2866** [comment]
Try first to flash the known working Webcam precompiled one from here https://tasmota.github.io/install/

**Reply by groovetronic911** [reply]
> Try first to flash the known working Webcam precompiled one from here https://tasmota.github.io/install/

You mean tasmota32-webcam.bin from http://ota.tasmota.com/tasmota32/release/ ?

I already tried it - without success.

**Reply by Jason2866** [reply]
Done with`esptool.py write_flash 0x0 tasmota32-webcam.factory.bin`?

**Reply by groovetronic911** [reply]
> Done with`esptool.py write_flash 0x0 tasmota32-webcam.factory.bin`?

I did it with nodemcu-pyflasher.
I'll give it a try!

**Reply by Jason2866** [reply]
Using NodeMCU-PyFlasher with the factory images can end in not booting device.
Pyflasher maybe detects not correctly the factory image and tries to change flash settings.
This damages the firmware image. ONLY use the recommended tools mentioned in Tasmota docu
NodeMCU-Pyflasher is not recommended anyways. It uses outdated esptool.py under the hood and  flashing the newer ESP32 MCUs is not working reliable. The tool needs maintain work!

**Reply by groovetronic911** [reply]
No success. I tried different tasmota32-bins with esptool. I haven't been able to connect to the own tasmota-Wifi - there was no own Wifi. Now I will wait for a new module. Perhaps my module is not ok.

**Reply by groovetronic911** [reply]
So...I have a new ESP32-CAM with ESP32-CAM-MB. With my own compiled file, there is Wifi and full function. 
On my ESP32-wrover there is no wifi, so I can do nothing.

I also made an installation via https://tasmota.github.io/install/ on both modules - ESP32-CAM works very fine - wrover doesn't work.

I think my wrover module is not working correctly or it is not the right module for m
